src/gpg.ts

```diff
@@ -0,0 +1,558 @@
+const getCipherAlgorithms: Fig.Generator = {
+  script: "gpg --version",
+  postProcess: (out, context) => {
+    // Get the substring of digests, remove whitespace and split by commas
+    let cyphers = out.substring(out.indexOf("Cypher: ") + 8, out.indexOf("Hash: "))
+    cyphers = cyphers.replace(/\s+/g, '')
+    cyphers = cyphers.split(",")
+
+    return cyphers.map(cypher => { name: cypher });
+  },
+};
+
+const getDigestAlgorithms: Fig.Generator = {
+  script: "gpg --version",
+  postProcess: (out, context) => {
+    // Get the substring of cyphers, remove whitespace and split by commas
+    let digests = out.substring(out.indexOf("Hash: ") + 8, out.indexOf("Compression: "))
+    digests = cyphers.replace(/\s+/g, '')
+    digests = cyphers.split(",")
+
+    return digests.map(digest => { name: digest });
+  },
+};
+
+const completionSpec: Fig.Spec = {
+  name: "gpg",
+  description: "Encryption and signing tool",
+  options: [
+    {
+      name: "--homedir",
+      description: "Set the name of the home directory",
+      args: { name: "directory", template: "folders" },
+    },
+    {
+      name: "--options",
+      description: "Read options from file",
+      args: { name: "file", template: "filepaths" },
+    },
+    { name: ["-a", "--armour"], description: "Create ASCII armored output" },
+    {
+      name: ["-o", "--output"],
+      description: "Write output to file",
+      args: { name: "file", template: "filepaths" },
+    },
+    {
+      name: ["-u", "--local-user"],
+      description: "Use name as the user ID to sign",
+      args: { name: "name" },
+    },
+    {
+      name: "--default-key",
+      description: "Use name as default user ID for signatures",
+      args: { name: "name" },
+    },
+    {
+      name: ["-r", "--recipient"],
+      description: "Encrypt for user id name",
+      args: { name: "name" },
+    },
+    {
+      name: "--default-recipient",
+      description: "Use name as default recipient",
+      args: { name: "name" },
+    },
+    { name: "--default-recipient-self", description: "Use the default key as default recipient" },
+    { name: "--no-default-recipient", description: "Reset --default-recipient and --default-recipient-self" },
```

Comment:
This option is exclusive on its counterpart and vice versa:
```suggestion
    { name: "--no-default-recipient", exclusiveOn: ["--default-recipient", "--default-recipient-self"], description: "Reset --default-recipient and --default-recipient-self" },
```